.evergreen/download-mongodb.sh

```diff
@@ -56,7 +56,7 @@ get_mongodb_download_url_for ()
    VERSION_MONGOSH="2.1.1"
    # Set VERSION_RAPID to the latest rapid release each quarter.
    VERSION_RAPID="7.3.1"
-   VERSION_80="v8.0-latest"
+   VERSION_80="8.0.0-rc14"
```

Comment:
we're up to rc18 now: https://jira.mongodb.org/browse/STAR-5964

Reply:
lol and the link changed yet again